python/lsst/ts/cbp/component.py

```diff
@@ -257,16 +262,15 @@ async def connect(self):
             await self.client.start_task
         except Exception:
             self.log.exception("Connection failed.")
-            await self.fault(code=2, report="Connection failed.")
+            await self.csc.fault(code=2, report="Connection failed.")
```

Comment:
I just noticed that the `CBPCSC` and `CBPComponent` hold each other as a circular dependency. This is not good. Maybe consider to have a new ticket to decouple them? I would suggest to have the `CBPCSC` as the highest one and remove the knowledge of `CBPCSC` from `CBPComponent`.